pkg/operator/ceph/controller/spec.go

```diff
@@ -456,3 +463,52 @@ func StoredLogAndCrashVolumeMount(varLogCephDir, varLibCephCrashDir string) []v1
 		},
 	}
 }
+
+// GenerateLivenessProbeExecDaemon makes sure a daemon has a socket and that it can be called and returns 0
+func GenerateLivenessProbeExecDaemon(daemonType, daemonID string) *v1.Probe {
+	confDaemon := getDaemonConfig(daemonType, daemonID)
+
+	return &v1.Probe{
+		Handler: v1.Handler{
+			Exec: &v1.ExecAction{
+				// Run with env -i to clean env variables in the exec context
+				// This avoids conflict with the CEPH_ARGS env
+				//
+				// Example:
+				// env -i sh -c ceph --admin-daemon /run/ceph/ceph-osd.0.asok status
+				Command: []string{
+					"env",
+					"-i",
+					"sh",
+					"-c",
+					fmt.Sprintf("ceph --admin-daemon %s %s", confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand()),
+				},
+			},
+		},
+		InitialDelaySeconds: 10,
+	}
+}
+
+func getDaemonConfig(daemonType, daemonID string) *daemonConfig {
+	return &daemonConfig{
+		daemonType: string(daemonType),
+		daemonID:   daemonID,
+	}
+}
+
+func (c *daemonConfig) buildSocketName() string {
+	return fmt.Sprintf("ceph-%s.%s.asok", c.daemonType, c.daemonID)
+}
+
+func (c *daemonConfig) buildSocketPath() string {
+	return path.Join(daemonSocketDir, c.buildSocketName())
+}
+
+func (c *daemonConfig) buildAdminSocketCommand() string {
```

Comment:
Does connecting to the admin socket require the cephx key? Or it's just the local socket so the keys aren't necessary?

Reply:
The socket is a network socket, it's to communicate with other daemons.  In this case, keys are not necessary, it's a local call to the daemon process.